### PR TITLE
[e2e] Make kind verbose

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
[e2e] Make kind verbose

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/e2e-test.sh [hunk 1]
```diff
@@ -34,15 +34,18 @@ function cleanup {
         kubectl logs -n kube-system kube-controller-manager-kind-control-plane > $ARTIFACTS/kube-controller-manager.log || true
         kubectl logs -n kueue-system deployment/kueue-controller-manager > $ARTIFACTS/kueue-controller-manager.log || true
         kubectl describe pods -n kueue-system > $ARTIFACTS/kueue-system-pods.log || true
-        $KIND delete cluster --name $KIND_CLUSTER_NAME || { echo "You need to run make kind-image-build before this script"; exit -1; }
+        $KIND delete cluster --name $KIND_CLUSTER_NAME -v 5
```
Comment (tenzen-y) [inline hack/e2e-test.sh:37]: Why do we need to increase verbose? Do you have any cases you want to increase this verbose?

Comment (tenzen-y) [reply hack/e2e-test.sh:37]: My concern is to overlook important logs by this increase.

Comment (trasc) [reply hack/e2e-test.sh:37]: there war some cases when the kind cluster did not start. https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/913/pull-kueue-test-e2e-main-1-25/1674051550193913856

Comment (alculquicondor) [reply hack/e2e-test.sh:37]: I guess it's not worth during delete, just create?

File: hack/e2e-test.sh [hunk 2]
```diff
@@ -34,15 +34,18 @@ function cleanup {
...
     fi
     (cd config/components/manager && $KUSTOMIZE edit set image controller=gcr.io/k8s-staging-kueue/kueue:main)
 }
 
 function startup {
     if [ $CREATE_KIND_CLUSTER == 'true' ]
     then
-        $KIND create cluster --name $KIND_CLUSTER_NAME --image $E2E_KIND_VERSION --config $SOURCE_DIR/kind-cluster.yaml --wait 1m
+        $KIND create cluster --name $KIND_CLUSTER_NAME --image $E2E_KIND_VERSION --config $SOURCE_DIR/kind-cluster.yaml --wait 1m -v 5
```
Comment (alculquicondor) [inline hack/e2e-test.sh:45]: Can we start with something like v2?

File: hack/e2e-test.sh (hunk 1 above)
Comment (trasc) [reply hack/e2e-test.sh]: done

File: hack/e2e-test.sh (hunk 2 above)
Comment (alculquicondor) [reply hack/e2e-test.sh:45]: let's output to another file.... just to reduce clutter from the main logs

Comment (tenzen-y) [reply hack/e2e-test.sh:45]: > let's output to another file.... just to reduce clutter from the main logs

+1

Otherwise LGTM

Comment (trasc) [reply hack/e2e-test.sh]: done, redirect to file, `cat` the file if create fails

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**